interactive/views.py

```diff
@@ -225,6 +225,18 @@ def dispatch(self, request, *args, **kwargs):
         if not self.analysis_request.report:
             raise Http404
 
+        # we can only edit reports if they don't have any publish requests or
+        # the latest one is rejected
+        latest_publish_request = self.analysis_request.report.publish_requests.order_by(
+            "-created_at"
+        ).first()
+        if latest_publish_request and (
+            latest_publish_request.decision is None
+            or latest_publish_request.decision
+            == ReportPublishRequest.Decisions.APPROVED
+        ):
+            raise PermissionDenied
```

Comment:
Is `PermissionDenied` the right error? It would be nice if the error indicated that it's the state of the resource that's wrong, not the permissions of the user.

Reply:
Maybe…  We've disabled the button used to get here so this is to catch users who go directly to the URL. And now that I write it out, of course someone is going to be on that page or have a URL somehow.  I'll put a template together for this 👍